ggml/src/ggml-cpu/vec.cpp

```diff
@@ -404,6 +404,70 @@ void ggml_vec_swiglu_f32(const int n, float * y, const float * x, const float *
     }
 }
 
+ggml_float ggml_vec_cvar_f32(const int n, float * y, const float * x, const float mean) {
+  int i = 0;
+  ggml_float sum = 0;
+#if defined(__AVX512F__) && defined(__AVX512DQ__)
+    for (; i + 15 < n; i += 16) {
+        __m512 val = _mm512_sub_ps(_mm512_loadu_ps(x + i),
+                                   _mm512_set1_ps(mean));
+        _mm512_storeu_ps(y + i, val);
+        sum += (ggml_float)_mm512_reduce_add_ps(_mm512_mul_ps(val, val));
+    }
+#elif defined(__AVX2__) && defined(__FMA__)
+    for (; i + 7 < n; i += 8) {
```

Comment:
Due to the `#elif`, this branch will never be used if there is AVX512 (same for similar branches below). The loop structure however suggests that the intention is to process the leftover element in smaller batches.

Reply:
The leftover elements are processed  after the final #endif at line 461. It tricked me too when I read the code of ggml_vec_soft_max_f32 (the same pattern is present there), but I think it's correct, though quite terse.

Reply:
It could still be advantageous to process the remaining items in groups using the smaller vector sizes from AVX2 and SSE. This is done in `ggml_cpu_fp32_to_fp16`, for example. There are probably not many cases where this would make a measurable difference in performance, but this seems like an odd way to write the loops if that was not the original intention.

Reply:
@duduta Add a small "TODO" comment referencing this discussion so we don't forget to improve this.